hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientRingbufferProxy.java

```diff
@@ -0,0 +1,248 @@
+    }
+
+    @Override
+    public ICompletableFuture<Long> addAllAsync(Collection<? extends E> collection, OverflowPolicy overflowPolicy) {
+        checkNotNull(collection, "collection can't be null");
+        checkNotNull(overflowPolicy, "overflowPolicy can't be null");
+        checkFalse(collection.isEmpty(), "collection can't be empty");
+        checkTrue(collection.size() <= MAX_BATCH_SIZE, "collection can't be larger than " + MAX_BATCH_SIZE);
+
+        final List<Data> valueList = new ArrayList<Data>(collection.size());
+        for (E e : collection) {
+            throwExceptionIfNull(e);
+            valueList.add(toData(e));
+        }
+
+        ClientMessage request = RingbufferAddAllCodec.encodeRequest(getName(), valueList, overflowPolicy.getId());
+        request.setPartitionId(partitionId);
+
+        try {
+            ClientInvocationFuture invocationFuture = new ClientInvocation(getClient(), request).invoke();
+            return new ClientDelegatingFuture<Long>(
+                    invocationFuture,
+                    getContext().getSerializationService(),
+                    ADD_ALL_ASYNC_RESPONSE_DECODER);
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        }
+    }
+
+    @Override
+    public ICompletableFuture<ReadResultSet<E>> readManyAsync(long startSequence, int minCount,
+                                                              int maxCount, IFunction<E, Boolean> filter) {
+
+        checkSequence(startSequence);
+        checkNotNegative(minCount, "minCount can't be smaller than 0");
+        checkTrue(maxCount >= minCount, "maxCount should be equal or larger than minCount");
+        checkTrue(minCount <= capacity(), "the minCount should be smaller than or equal to the capacity");
+        checkTrue(maxCount <= MAX_BATCH_SIZE, "maxCount can't be larger than " + MAX_BATCH_SIZE);
+
+        ClientMessage request = RingbufferReadManyCodec.encodeRequest(
+                getName(),
+                startSequence,
+                minCount,
+                maxCount,
+                toData(filter));
+        request.setPartitionId(partitionId);
+
+        try {
+            ClientInvocationFuture invocationFuture = new ClientInvocation(getClient(), request).invoke();
+            return new ClientDelegatingFuture<ReadResultSet<E>>(
+                    invocationFuture,
+                    getContext().getSerializationService(),
+                    readManyAsyncResponseDecoder);
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        }
+    }
+
+    private static void checkSequence(long sequence) {
+        if (sequence < 0) {
+            throw new IllegalArgumentException("sequence can't be smaller than 0, but was: " + sequence);
+        }
+    }
+
+    @Override
+    protected <T> T invoke(ClientMessage req) {
+        req.setPartitionId(partitionId);
+        return super.invoke(req);
```

Comment:
This will be invoked on a random node, and  I think intention as to send to request to owner of partition. 
This will call the 
```
 new ClientInvocation(getClient(), clientMessage).invoke();
```
We need the send via following
```
 new ClientInvocation(getClient(), clientMessage, partitionId).invoke();
```
And setting partition id on message will be done internally by client invocation system.

I think it make sense to add a method as following to ClientProxy for these kind of usages

```
  protected <T> T invoke(ClientMessage clientMessage, int partitionId) {
        try {
            final Future future = new ClientInvocation(getClient(), clientMessage, partitionId).invoke();
            return (T) future.get();
        } catch (Exception e) {
            throw ExceptionUtil.rethrow(e);
        }
    }
```